src/fmu/dataio/_model/fields.py

```diff
@@ -134,6 +134,12 @@ class Parameters(RootModel):
     root: Dict[str, Union[Parameters, int, float, str]]
     """A dictionary representing parameters as-is from parameters.txt."""
 
+    def __iter__(self) -> Dict[str, Union[Parameters, int, float, str]]:
+        return iter(self.root)
```

Comment:
This type annotation is not quite right. `iter()` creates an iterator over the `Dict[str, Union[Parameters, int, float, str]]` so it won't return the dict, and by default iterating over the dict iterates over the keys of the dict.

So it should be:

```suggestion
    def __iter__(self) -> Iterator[str]:
        return iter(self.root)
```

because we have the keys as strings: `Dict[str, ...]`

Reply:
Thanks, I was a little bit too quick ...